cmake/unity.cmake

```
@@ -0,0 +1,12 @@
+function(TT_ENABLE_UNITY_BUILD TARGET)
+    if(TT_UNITY_BUILDS)
+        set_target_properties(
+            ${TARGET}
+            PROPERTIES
+                UNITY_BUILD
+                    ON
+                UNITY_BUILD_UNIQUE_ID
+                    "CMAKE_UNIQUE_NAMESPACE"
+        )
+    endif()
+endfunction()
```

Here's what I'm trying to suggest:

```suggestion
set(CMAKE_UNITY_BUILD_UNIQUE_ID CMAKE_UNIQUE_NAMESPACE)
function(TT_ENABLE_UNITY_BUILD TARGET)
    if(TT_UNITY_BUILDS)
        set_target_properties(
            ${TARGET}
            PROPERTIES
                UNITY_BUILD
                    ON
        )
    endif()
endfunction()
```

---

I'm a bit hesitant to do a global `set(CMAKE_UNITY_BUILD_UNIQUE_ID CMAKE_UNIQUE_NAMESPACE)` in case some of our dependencies may decide to use a different one

---

If they use it, they need to define it, which will be within their scope.  So then in their scope they'll have their value and when returning to our scope we'll be back to our value.

Just like using `int idx = foo;` and not worrying that we may call a function who also defines a var named `idx`.  It's all good.

---

I agree, using a global `CMAKE_UNITY_BUILD_UNIQUE_ID` shouldn't cause any problems.
But I would still prefer to specify it per target within a function, this seems like a cleaner approach to me. Also, I'm a bit concerned if older cmakes which don't recognize `CMAKE_UNITY_BUILD_UNIQUE_ID` would give you an unnecessary warning about it.

---

There won't be warnings, because it's just a variable.  Like any custom variable.  And it won't be looked at unless BUILD_UNITY is truth-y.

But it's a minor nit.  Let's not spend time on this point.  That just delays getting faster builds in place.